src/DurableTask.SqlServer/DTUtils.cs

```diff
@@ -180,5 +200,52 @@ public static bool HasPayload(HistoryEvent e)
                 _ => null,
             };
         }
+
+        public static bool TryDeserializeFailureDetails(string jsonText, out FailureDetails? failureDetails)
+        {
+            try
+            {
+                failureDetails = DeserializeFromJson<FailureDetails>(jsonText);
+
+                // Depending on how the TaskHubWorker is configured, the text might be a FailureDetails JSON object
```

Comment:
Just curious, can you share a link to the code where the `ErrorPropagationMode` is set to the default of `ErrorPropagationMode.UseFailureDetails`?

Reply:
In this project, the default is set up in TestService.cs, line 68. The default mode can be set in the constructor parameter. For Orchestrations.cs, we don't specify anything, which causes the default value to be `UseFailureDetails`. Only in LegacyErrorPropagation.cs do we use the `SerializeExceptions` mode.

FWIW, the property is defined here: [`TaskHubWorker.ErrorPropagationMode`](https://github.com/Azure/durabletask/blob/6dc7134ceee4cec9073b86d5bb70da0cd4ca6731/src/DurableTask.Core/TaskHubWorker.cs#L130-L145). We don't set the default value explicitly, but enum properties default to zero, which in this case is [`SerializeExceptions`](https://github.com/Azure/durabletask/blob/6dc7134ceee4cec9073b86d5bb70da0cd4ca6731/src/DurableTask.Core/ErrorPropagationMode.cs#L28). This is the default because that was always the original behavior of the Durable Task Framework and changing the default behavior could break in-flight orchestrations for existing apps. It was much later that we added support for `UseFailureDetails`.

Reply:
Not sure if this is what you were asking, but we also set it in the .NET Isolated SDK [here](https://github.com/microsoft/durabletask-dotnet/blob/e2f329258d1e02a762911bd89274a9a584b6ec04/src/Worker/Grpc/GrpcDurableTaskWorker.Processor.cs#L213-L217).

Java assumes this behavior as well, but doesn't require any explicit configuration since Java doesn't rely on the .NET implementation of `TaskHubWorker`. Basically, this is the _only_ option for Java.

Reply:
Thanks for the explanation! I understand it a lot better now.